
* Работа для агенства "Digital Scouting"
* Данные по зарплатам игроков примерно 30 ведущих футбольных лиг.

# Содержание

1. [Подготовим данные и сделаем первые выводы](#view1)
2. [Зарплатный рейтинг по возврастам](#view5)
  - 2.1 [Итоговая таблица по возрастам](#age1)
3. [Зарплатный рейтинг по позициям](#view4)
  - 3.1 [Итоговая таблица по позициям](#poz1)
4. [Национальный рейтинг](#naz1)
  - 4.1 [Национальный рейтинг итоговая таблица](#nazf)
5. [Национальный рейтинг по зп](#naz2)
  - 5.1 [Национальный рейтинг по игрокам итоговая таблица](#nazf2)
6. [Рейтинг клубов по ЗП ведомости](#view2) 

 - 6.1 [Итог таблица перцентили](#final1)
 - 6.2 [Итог таблица по игрокам](#final2)
 
7. [Рейтинг лиг по зп ведомости](#view3)
  - 7.1 [Итоговая таблица по рейтингу лиг](#zp1)


8. [Вывод](#vivod1)

## 1.Подготовим данные и сделаем первые выводы<a name="view1"></a>

In [1]:
import pandas as pd
import json
from IPython.display import display
import sklearn
from sklearn.cluster import KMeans
import numpy as np
import math
import matplotlib.pyplot as plt
import os

pd.set_option('display.max_colwidth', None)
pd.set_option('display.float_format', '{:.2f}'.format)

In [2]:
data = pd.read_csv('data_capology.csv',sep = ",",low_memory=False)
data = data.replace("\'", '\"')

In [3]:
#увеличим вывод строк
pd.options.display.max_rows = 1000

In [4]:
display(data.head(12))
display(data.shape)

player  net_pw    net_py position  age      country  \
0        Kylian Mbappé  977432  50826480        F   23       France   
1         Lionel Messi  684203  35578536        F   34    Argentina   
2               Neymar  606008  31512418        F   29       Brazil   
3    Cristiano Ronaldo  579644  30141507        F   36     Portugal   
4          Gareth Bale  319034  16589763        F   32        Wales   
5          Eden Hazard  293230  15247944        F   31      Belgium   
6         Sergio Ramos  293230  15247944        D   35        Spain   
7      Frenkie de Jong  273681  14231414        M   24  Netherlands   
8      Kevin De Bruyne  265245  13792754        F   30      Belgium   
9         David de Gea  248783  12936735        K   31        Spain   
10  Robert Lewandowski  238298  12391496        F   33       Poland   
11        Jadon Sancho  232090  12068660        F   21      England   

                 club          league  
0                 PSG         Ligue 1  
1                 PSG         Ligue 1  
2                 PSG         Ligue 1  
3   Manchester United  Premier League  
4         Real Madrid         La Liga  
5         Real Madrid         La Liga  
6                 PSG         Ligue 1  
7           Barcelona         La Liga  
8     Manchester City  Premier League  
9   Manchester United  Premier League  
10      Bayern Munich    1.Bundesliga  
11  Manchester United  Premier League

(14291, 8)

In [5]:
#проверим тип данных
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14291 entries, 0 to 14290
Data columns (total 8 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   player    14291 non-null  object
 1   net_pw    14291 non-null  object
 2   net_py    14291 non-null  int64 
 3   position  14291 non-null  object
 4   age       14291 non-null  int64 
 5   country   14291 non-null  object
 6   club      14291 non-null  object
 7   league    14291 non-null  object
dtypes: int64(2), object(6)
memory usage: 893.3+ KB


In [6]:
# с помощью лямб функции поменяем тип данных с object  на  float для того что бы произвести расчеты
#data['net_py'] = data['net_py'].apply(lambda x: x.replace('$', '').replace(',', '')).astype('float')
#data['net_pw'] = data['net_pw'].apply(lambda x: x.replace('$', '').replace(',', '')).astype('float')


## 2.Зарплатный рейтинг по возврастам <a name="view5"></a>
Возрастная категория - Максимальная зп - Перцентиль 90 -Перцентиль 80 -Перцентиль 60 -Перцентиль 40
- до 20					
21-23					
24-26					
27-29					
30-32					
от 33					
					

In [7]:
def age_group(age):
    if age <= 20:
        return('до 20')
    if 21 <= age <= 23:
        return('21-23')
    if 24 <= age <= 26:
        return('24-26')
    if 27 <= age <= 29:
        return('27-29')
    if 30 <= age <= 32:
        return('30-32')
    if age > 30:
        return('от 33')
    return 

In [8]:
data['age_group'] = data['age'].apply(age_group)
print(data['age_group'].value_counts())

21-23    3364
24-26    3205
27-29    2600
до 20    2123
30-32    1794
от 33    1205
Name: age_group, dtype: int64


In [9]:
age = data.groupby(['age_group']).agg({'net_py':'max'})
age.sort_values(by='net_py',ascending = False,inplace=True)
display(age.shape)

(6, 1)

In [10]:
perage_40 = data.groupby('age_group')['net_py'].quantile(0.4)
perage_60 = data.groupby('age_group')['net_py'].quantile(0.6)
perage_80 = data.groupby('age_group')['net_py'].quantile(0.8)
perage_90 = data.groupby('age_group')['net_py'].quantile(0.9)

data_perc_age = pd.DataFrame()
data_perc_age['percentil40'] = perage_40
data_perc_age['percentil60'] = perage_60
data_perc_age['percentil80'] = perage_80
data_perc_age['percentil90'] = perage_90
display(data_perc_age.shape)

(6, 4)

In [11]:
#соеденим таблицы 
rating_age = age.merge(data_perc_age ,left_on='age_group' ,right_on = 'age_group' )
display(rating_age.shape)

(6, 5)

## 2.1 Итоговая таблица по возрастам<a name="age1"></a>

In [12]:
#Рейтинг зп по возрастам
final_age = rating_age.rename({'net_py':'макс зп','percentil40':'перцентиль40','percentil60':'перцентиль60','percentil80':'перцентиль80','percentil90':'перцентиль90'}, axis=1)
final_age = final_age.sort_values(by=['макс зп'],ascending=False).reset_index()
final_age.groupby('age_group')
display(final_age.shape)
final_age

(6, 6)

age_group   макс зп  перцентиль40  перцентиль60  перцентиль80  перцентиль90
0     21-23  50826480      48226.00      91488.00     213471.00     437108.00
1     от 33  35578536     182975.00     365951.00     753698.80    1502678.40
2     27-29  31512418     142314.00     294794.00     640414.00    1266414.90
3     30-32  16589763     172810.00     374751.00     823389.00    1493009.30
4     24-26  14231414      91488.00     200000.00     461520.80    1002298.00
5     до 20   6810748      24113.00      40661.00      91488.00     193141.00

In [13]:
data.sort_values(by='net_py',ascending = False)
data.head(541)

player   net_pw    net_py position  age  \
0                Kylian Mbappé   977432  50826480        F   23   
1                 Lionel Messi   684203  35578536        F   34   
2                       Neymar   606008  31512418        F   29   
3            Cristiano Ronaldo   579644  30141507        F   36   
4                  Gareth Bale   319034  16589763        F   32   
5                  Eden Hazard   293230  15247944        F   31   
6                 Sergio Ramos   293230  15247944        D   35   
7              Frenkie de Jong   273681  14231414        M   24   
8              Kevin De Bruyne   265245  13792754        F   30   
9                 David de Gea   248783  12936735        K   31   
10          Robert Lewandowski   238298  12391496        F   33   
11                Jadon Sancho   232090  12068660        F   21   
12              Raphaël Varane   225366  11719018        D   28   
13               Karim Benzema   225200  11710421        F   34   
14                Manuel Neuer   217576  11313974        K   36   
15               Romelu Lukaku   215628  11212641        F   28   
16               Thomas Müller   212494  11049677        F   32   
17                 David Alaba   211125  10978520        D   29   
18                  Leroy Sané   207216  10775214        F   26   
19             Sergio Busquets   206434  10734553        M   33   
20                     Marcelo   205847  10704057        D   33   
21                  Toni Kroos   205847  10704057        M   32   
22           Philippe Coutinho   205261  10673561        F   29   
23              Joshua Kimmich   202133  10510916        M   26   
24               Jack Grealish   198934  10344565        F   26   
25             Raheem Sterling   198934  10344565        F   27   
26           Antoine Griezmann   195486  10165296        F   30   
27            Presnel Kimpembe   195486  10165296        D   26   
28        Gianluigi Donnarumma   195486  10165296        K   22   
29                  Jordi Alba   195486  10165296        D   32   
30                   Jan Oblak   195095  10144965        K   29   
31                  Paul Pogba   192210   9994924        M   28   
32                N'Golo Kanté   191978   9982867        M   30   
33             Ousmane Dembélé   187667   9758684        F   24   
34             Lucas Hernández   186494   9697692        D   25   
35               Leon Goretzka   186494   9697692        M   26   
36                 Luka Modric   184735   9606205        M   36   
37                 Timo Werner   180385   9380037        F   25   
38              Kingsley Coman   176133   9158932        F   25   
39                 John Stones   165778   8620471        D   27   
40             Anthony Martial   165778   8620471        F   25   
41              Edinson Cavani   165778   8620471        F   34   
42   Pierre-Emerick Aubameyang   165778   8620471        F   32   
43             Bruno Fernandes   159054   8270830        F   27   
44               Achraf Hakimi   156389   8132237        D   23   
45            Matthijs de Ligt   156389   8132237        D   22   
46                  Marquinhos   154825   8050914        D   27   
47              Marco Verratti   154825   8050914        M   29   
48                        Koke   150134   7806947        M   30   
49           Christian Eriksen   146615   7623972        F   29   
50                 Luis Suárez   146615   7623972        F   35   
51             Virgil van Dijk   145839   7583603        D   30   
52                Paulo Dybala   142705   7420666        F   28   
53              Ángel Di María   141923   7380005        F   33   
54                Aaron Ramsey   136841   7115707        M   30   
55               Adrien Rabiot   136841   7115707        M   26   
56              Alexis Sánchez   136841   7115707        F   33   
57              Dusan Vlahovic   136841   7115707        F   22   
58         Georginio Wijnaldum   136841   7115707        M   31   
59  

 ## Вывод:  
 * Игроки 21-23 лет имеют наивысшую зп в своих клубах с достаточно сильным отрывом. 
 * Наименее оплачеваемые игроки оказались в возрасте до 20 лет, что в целом логично.
 

## 3.Зарплатный рейтинг по позициям<a name="view4"></a>
Позиция - Максимальная зп - Перцентиль 90 -Перцентиль 80 -Перцентиль 60 -Перцентиль 40

In [14]:
pos = data.groupby(['position']).agg({'net_py':'max'})
pos.sort_values(by='net_py',ascending = False,inplace=True)
display(pos.shape)

(4, 1)

In [15]:
perpos_40 = data.groupby('position')['net_py'].quantile(0.4)
perpos_60 = data.groupby('position')['net_py'].quantile(0.6)
perpos_80 = data.groupby('position')['net_py'].quantile(0.8)
perpos_90 = data.groupby('position')['net_py'].quantile(0.9)

data_perc_pos = pd.DataFrame()
data_perc_pos['percentil40'] = perpos_40
data_perc_pos['percentil60'] = perpos_60
data_perc_pos['percentil80'] = perpos_80
data_perc_pos['percentil90'] = perpos_90
display(data_perc_pos.shape)

(4, 4)

In [16]:
#соеденим таблицы 
rating_pos = pos.merge(data_perc_pos ,left_on='position' ,right_on = 'position' )
display(rating_pos.shape)

(4, 5)

## 3.1 Итоговая таблица по позициям <a name="poz1"></a>

In [17]:
#Рейтинг зп по позициям
final_pos = rating_pos.rename({'position':'позиция','net_py':'макс зп','percentil40':'перцентиль40','percentil60':'перцентиль60','percentil80':'перцентиль80','percentil90':'перцентиль90'}, axis=1)
final_pos = final_pos.sort_values(by=['макс зп'],ascending=False).reset_index()
final_pos.groupby('position')
display(final_pos.shape)
final_pos

(4, 6)

position   макс зп  перцентиль40  перцентиль60  перцентиль80  перцентиль90
0        F  50826480      80000.00     180849.00     490536.00    1016530.00
1        D  15247944      81322.00     172810.00     437108.00     813224.00
2        M  14231414      72340.00     180000.00     478215.20     914877.00
3        K  12936735      48226.00     101653.00     304959.00     609918.00

## 4. Национальный рейтинг <a name="naz1"></a>

* Национальность * Кол-во игроков в выборке * Зарабатывают больше 2 000 000 * Зарабатывают больше 1 000 000	Зарабатывают больше 600 000 *Зарабатывают больше 400 000 * Зарабатывают больше 200 000	* Зарабатывают больше 100 000

In [18]:
def player_group(net_py):
    if net_py >= 2000000:
         return('2000000')
    if net_py >= 1000000:
         return('1000000')
    if net_py >= 600000:
         return('600000')
    if net_py >= 400000:
         return('400000')
    if net_py >= 200000:
         return('200000')
    if net_py >= 100000:
         return('100000')
    return 

In [19]:
data['player_group'] = data['net_py'].apply(player_group)

In [20]:
dtt = data[['country','net_py','player_group']]
dtt.sort_values(by='net_py',ascending = False)
dtt.head(552)

country    net_py player_group
0                              France  50826480      2000000
1                           Argentina  35578536      2000000
2                              Brazil  31512418      2000000
3                            Portugal  30141507      2000000
4                               Wales  16589763      2000000
5                             Belgium  15247944      2000000
6                               Spain  15247944      2000000
7                         Netherlands  14231414      2000000
8                             Belgium  13792754      2000000
9                               Spain  12936735      2000000
10                             Poland  12391496      2000000
11                            England  12068660      2000000
12                             France  11719018      2000000
13                             France  11710421      2000000
14                            Germany  11313974      2000000
15                            Belgium  11212641      2000000
16                            Germany  11049677      2000000
17                            Austria  10978520      2000000
18                            Germany  10775214      2000000
19                              Spain  10734553      2000000
20                             Brazil  10704057      2000000
21                            Germany  10704057      2000000
22                             Brazil  10673561      2000000
23                            Germany  10510916      2000000
24                            England  10344565      2000000
25                            England  10344565      2000000
26                             France  10165296      2000000
27                             France  10165296      2000000
28                              Italy  10165296      2000000
29                              Spain  10165296      2000000
30                           Slovenia  10144965      2000000
31                             France   9994924      2000000
32                             France   9982867      2000000
33                             France   9758684      2000000
34                             France   9697692      2000000
35                            Germany   9697692      2000000
36                            Croatia   9606205      2000000
37                            Germany   9380037      2000000
38                             France   9158932      2000000
39                            England   8620471      2000000
40                             France   8620471      2000000
41                            Uruguay   8620471      2000000
42                              Gabon   8620471      2000000
43                           Portugal   8270830      2000000
44                            Morocco   8132237      2000000
45                        Netherlands   8132237      2000000
46                             Brazil   8050914      2000000
47                              Italy   8050914      2000000
48                              Spain   7806947      2000000
49                            Denmark   7623972      2000000
50                            Uruguay   7623972      2000000
51                        Netherlands   7583603      2000000
52                          Argentina   7420666      2000000
53                          Argentina   7380005      2000000
54                              Wales   7115707      2000000
55                             France   7115707      2000000
56                              Chile   7115707      2000000
57                             Serbia   7115707      2000000
58                        Netherlands   7115707      2000000
59                           Portugal   7115707      2000000
60                             Sweden   7115707      2000000
61                             Brazil   7075046      2000000
62                             France   6902236      2000000
63                             France   6902236      2000000
64                            England   6896377      2000000
65  

In [21]:
zar = data.groupby(['country']).agg({'player':'count'})
zar.sort_values(by='player',ascending = False,inplace=True)
display(zar.shape)

(153, 1)

In [22]:
zar_test = data.groupby(['country']).agg({'player':'count','player_group':'count'})
zar_test.sort_values(by='player',ascending = False,inplace=True)
display(zar_test.shape)

(153, 2)

In [23]:
#zar_test_2 = data.groupby(['country','player_group','net_py']).agg({'player':'count'})
#zar_test_2.sort_values(by='player',ascending = False,inplace=True)
#display(zar_test_2.shape)
#zar_test_2

zar_test_2 = data.groupby(['country','player_group']).agg({'net_py':'count'})
#zar_test_2.sort_values(by='player',ascending = False,inplace=True)
display(zar_test_2.shape)

(543, 1)

In [24]:
gpoup_pivot = pd.pivot_table(zar_test_2,
               index="country",
               values="net_py",
               columns="player_group"              
               ).reset_index()

#right_pivot.sort_values(by='player_group',ascending = False)

gpoup_pivot=gpoup_pivot.fillna(0)
display(gpoup_pivot.shape)

(140, 7)

In [25]:
play = gpoup_pivot.merge(zar ,left_on='country' ,right_on = 'country')
display(play.shape)
play = play.sort_values(by='player',ascending = False)

(140, 8)

## 4.1. Национальный рейтинг итоговая табица<a name="nazf"></a>

In [26]:
play = play.rename({'country':'Cтрана','player':'Кол-во игроков в выборке'}, axis=1)
display(play.shape)
play.reset_index()
play.head()

(140, 8)

Cтрана  100000  1000000  200000  2000000  400000  600000  \
43     England  206.00    84.00  159.00    56.00   57.00   57.00   
15      Brazil  153.00    71.00  171.00    50.00   87.00   63.00   
47      France  164.00    41.00  102.00    55.00   47.00   45.00   
117      Spain  172.00    65.00  171.00    65.00   61.00   52.00   
4    Argentina  204.00    35.00  171.00    22.00   93.00   57.00   

     Кол-во игроков в выборке  
43                       1532  
15                       1078  
47                        926  
117                       883  
4                         860

In [27]:
play['more100_000'] = play[['100000','1000000','200000','2000000','400000','600000']].sum( axis=1)
play['more200_000'] = play[['1000000','200000','2000000','400000','600000']].sum( axis=1)
play['more400_000'] = play[['1000000','2000000','400000','600000']].sum( axis=1)
play['more600_000'] = play[['1000000','2000000','600000']].sum( axis=1)
play['more1_000_000'] = play[['1000000','2000000']].sum( axis=1)
play['more2_000_000'] = play[['2000000']].sum( axis=1)
play

Cтрана  100000  1000000  200000  2000000  \
43                            England  206.00    84.00  159.00    56.00   
15                             Brazil  153.00    71.00  171.00    50.00   
47                             France  164.00    41.00  102.00    55.00   
117                             Spain  172.00    65.00  171.00    65.00   
4                           Argentina  204.00    35.00  171.00    22.00   
26                           Colombia  129.00    11.00   93.00     3.00   
51                            Germany  155.00    43.00  127.00    34.00   
67                              Italy  115.00    56.00  130.00    36.00   
93                        Netherlands   58.00    25.00   46.00    16.00   
133                     United States   62.00     6.00   52.00     5.00   
130                            Turkey   54.00    12.00   73.00     5.00   
86                             Mexico   39.00     7.00   64.00     7.00   
105                          Portugal   44.00    11.00   43.00    24.00   
10                            Belgium   55.00    17.00   34.00    12.00   
7                             Austria   64.00    10.00   30.00     4.00   
109                          Scotland   25.00     9.00   26.00     2.00   
108                            Russia   31.00    29.00   45.00     5.00   
123                       Switzerland   45.00    11.00   31.00     4.00   
65                            Ireland   24.00     5.00   19.00     3.00   
134                           Uruguay   33.00     9.00   38.00     9.00   
110                           Senegal   27.00    11.00   17.00     8.00   
137                             Wales   21.00     7.00   11.00     3.00   
96                            Nigeria   11.00     6.00   13.00     3.00   
31                            Croatia   20.00    14.00   17.00    10.00   
30                      Cote d'Ivoire   14.00     7.00   18.00     8.00   
111                            Serbia   20.00     9.00   19.00     7.00   
38                            Denmark   19.00    11.00   23.00     9.00   
52                              Ghana   20.00     6.00   16.00     2.00   
90                            Morocco   15.00     5.00    6.00     4.00   
122                            Sweden   16.00     5.00   18.00     4.00   
104                            Poland   12.00     5.00   15.00     9.00   
101                          Paraguay   14.00     8.00   17.00     0.00   
20                             Canada    7.00     1.00    9.00     1.00   
136                         Venezuela   15.00     2.00    8.00     1.00   
99                             Norway    9.00     8.00   14.00     3.00   
19                           Cameroon   11.00     2.00   11.00     2.00   
98                   Northern Ireland    9.00     1.00    7.00     1.00   
69                              Japan   18.00     3.00   10.00     1.00   
82                               Mali   12.00     2.00    9.00     1.00   
1                             Algeria    6.00     6.00   10.00     6.00   
24                              Chile    9.00     5.00   11.00     3.00   
40                            Ecuador   13.00     4.00   17.00     1.00   
53                             Greece   10.00     1.00   14.00     2.00   
37   Democratic Republic of the Congo    8.00     3.00    5.00     1.00   
14                 Bosnia-Herzegovina    8.00     2.00   14.00     3.00   
114                          Slovenia    7.00     4.00    5.00     2.00   
0                             Albania    4.00     4.00    7.00     1.00   
68                            Jamaica    6.00     2.00    8.00     2.00   
6                           Australia   10.00     0.00    6.00     0.00   
35                     Czech Republic    6.00     7.00    4.00     1.00   
75                             Kosovo    8.00     4.00    7.00     1.00   
107                           Romania    5.00     7.00    7.00     0.00   
46                            Finland    7.00     1.00  

In [28]:
play_drop=play.drop(['100000','1000000','200000','2000000','400000','600000'], axis=1)
play_drop

Cтрана  Кол-во игроков в выборке  more100_000  \
43                            England                      1532       619.00   
15                             Brazil                      1078       595.00   
47                             France                       926       454.00   
117                             Spain                       883       586.00   
4                           Argentina                       860       582.00   
26                           Colombia                       838       280.00   
51                            Germany                       754       470.00   
67                              Italy                       714       451.00   
93                        Netherlands                       413       191.00   
133                     United States                       403       162.00   
130                            Turkey                       388       200.00   
86                             Mexico                       356       177.00   
105                          Portugal                       348       153.00   
10                            Belgium                       324       146.00   
7                             Austria                       301       129.00   
109                          Scotland                       286        84.00   
108                            Russia                       284       202.00   
123                       Switzerland                       200       115.00   
65                            Ireland                       174        66.00   
134                           Uruguay                       174       120.00   
110                           Senegal                       123        74.00   
137                             Wales                       122        54.00   
96                            Nigeria                       102        59.00   
31                            Croatia                       101        78.00   
30                      Cote d'Ivoire                        99        66.00   
111                            Serbia                        98        77.00   
38                            Denmark                        98        79.00   
52                              Ghana                        97        58.00   
90                            Morocco                        89        50.00   
122                            Sweden                        85        61.00   
104                            Poland                        79        59.00   
101                          Paraguay                        70        50.00   
20                             Canada                        68        23.00   
136                         Venezuela                        68        32.00   
99                             Norway                        65        50.00   
19                           Cameroon                        65        38.00   
98                   Northern Ireland                        63        25.00   
69                              Japan                        60        39.00   
82                               Mali                        60        30.00   
1                             Algeria                        57        32.00   
24                              Chile                        54        40.00   
40                            Ecuador                        53        47.00   
53                             Greece                        52        39.00   
37   Democratic Republic of the Congo                        50        29.00   
14                 Bosnia-Herzegovina                        49        38.00   
114                          Slovenia                        44        33.00   
0                             Albania                        40        24.00   
68                            Jamaica                        39        25.00   
6                           Australia                        39        19.00   
35                     Czech Re

In [29]:
teto = data.query('country == "England"')
teto =teto.sort_values(by='net_py',ascending = False)
teto.head(57).reset_index()

index                   player  net_pw    net_py position  age  country  \
0      11             Jadon Sancho  232090  12068660        F   21  England   
1      25          Raheem Sterling  198934  10344565        F   27  England   
2      24            Jack Grealish  198934  10344565        F   26  England   
3      39              John Stones  165778   8620471        D   27  England   
4      65          Marcus Rashford  132623   6896377        F   24  England   
5      64               Harry Kane  132623   6896377        F   28  England   
6      77             Ben Chilwell  125899   6546735        D   25  England   
7      78            Harry Maguire  125899   6546735        D   28  England   
8      83   Trent Alexander-Arnold  119407   6209151        D   23  England   
9     103                Luke Shaw   99467   5172283        D   26  England   
10    126          Kieran Trippier   95757   4979377        D   31  England   
11    128              Jamie Vardy   92743   4822641        F   35  England   
12    129         Jordan Henderson   92743   4822641        M   31  England   
13    127             James Milner   92743   4822641        M   36  England   
14    138            Tammy Abraham   87969   4574383        F   24  England   
15    158  Alex Oxlade-Chamberlain   79527   4135415        M   28  England   
16    159                Ben White   79527   4135415        D   24  England   
17    160       Callum Hudson-Odoi   79527   4135415        F   21  England   
18    161               Danny Ings   79527   4135415        F   29  England   
19    191           Chris Smalling   74285   3862812        D   32  England   
20    193           James Maddison   73035   3797830        F   25  England   
21    194              Kyle Walker   73035   3797830        D   31  England   
22    224                Dele Alli   66311   3448188        F   25  England   
23    225          Jordan Pickford   66311   3448188        K   27  England   
24    223           Dean Henderson   66311   3448188        K   24  England   
25    222         Danny Drinkwater   66311   3448188        M   31  England   
26    244             Jarrod Bowen   63761   3315566        F   25  England   
27    245             Ross Barkley   63761   3315566        M   28  England   
28    264             Adam Lallana   60051   3122660        F   33  England   
29    265        Aaron Wan-Bissaka   59587   3098547        D   24  England   
30    274              Mason Mount   58660   3050321        F   23  England   
31    313          Nathaniel Clyne   53559   2785075        D   30  England   
32    314             Fabian Delph   53095   2760962        M   32  England   
33    315             Tyrone Mings   53095   2760962        D   28  England   
34    347              Ben Godfrey   51009   2652453        D   24  England   
35    355            Ollie Watkins   49849   2592170        F   26  England   
36    357             Theo Walcott   49849   2592170        F   32  England   
37    356               Phil Jones   49849   2592170        D   29  England   
38    353                Joe Gomez   49849   2592170        D   24  England   
39    354          Mason Greenwood   49849   2592170        F   19  England   
40    352            Jesse Lingard   49849   2592170        F   29  England   
41    390                Eric Dier   47995   2495717        D   28  England   
42    391            Axel Tuanzebe   47503   2470167        D   24  England   
43    399           Fraser Forster   46372   2411321        K   33  England   
44    400            Jonjo Shelvey   46372   2411321        M   29  England   
45    433         Brandon Williams   43126   2242528        D   21  England   
46    434            Michael Keane   43126   2242528        D   29  England   
47    455       Nathaniel Phillips   42894   2230472        D   24  England   
48    478           Aaron Ramsdale   41039   2134019        K   23  England   
49    488               Ben Davies   39880   2073736  

## 5. Национальный рейтинг по зп <a name="naz2"></a>

Национальность	Наивысшая зарплата	Зарплата 10 игрока	Зарплата 30 игрока	Зарплата 50 игрока	Зарплата 100 игрока	Зарплата 200 игрока	Зарплата 300 игрока

In [30]:
zp = data.groupby(['country']).agg({'net_py':'max'})
zp.sort_values(by='net_py',ascending = False,inplace=True)
display(zp.shape)
zp.head(1)

(153, 1)

net_py
country          
France   50826480

In [31]:
filter_list=[10,30,50,100,200,300]

In [32]:
#data_coutry = data.copy()
#data_coutry['rank'] = data_coutry.groupby('country')['net_py'].rank(method='dense',ascending=False)
#data_coutry_filtered = data_coutry.query("rank == @filter_list")
#data_coutry_filtered.sort_values(by = 'rank', ascending = False)
#display(data_coutry_filtered.shape)
#data_coutry_filtered.head(305)


data['rank'] = data.groupby('country')['net_py'].rank(method='dense',ascending=False)
data_coutry_filtered = data.query("rank == @filter_list")
data_coutry_filtered.sort_values(by = 'rank', ascending = False)
display(data_coutry_filtered.shape)
data_coutry_filtered.head(305)

(305, 11)

player   net_pw   net_py position  age  \
40              Anthony Martial   165778  8620471        F   25   
81                Cesc Fàbregas   122375  6363475        M   34   
109                 Kai Havertz    99467  5172283        F   22   
132                    Casemiro    92465  4808185        M   29   
138               Tammy Abraham    87969  4574383        F   24   
246                      Vitolo    63338  3293556        F   32   
252             Rodrigo de Paul    62556  3252895        M   27   
259              Gelson Martins    61187  3181738        F   26   
261                Thomas Lemar    60992  3171572        F   26   
271          Leander Dendoncker    59587  3098547        M   26   
294             Patrick Cutrone    58255  3029258        F   24   
362             Steven Bergwijn    49849  2592170        F   24   
411                        Hulk    45962  2390000        F   35   
469                   Pau López    41443  2155043        K   27   
506                   Max Kruse    39293  2043224        F   33   
507               Fikayo Tomori    39097  2033059        D   24   
579        Jean-Philippe Mateta    35474  1844660        F   24   
582                  Salif Sané    35383  1839919        D   31   
609               Daler Kuzyaev    34015  1768762        M   29   
610            Georgiy Dzhikiya    34015  1768762        D   28   
611              Magomed Ozdoev    34015  1768762        M   29   
659              Thomas Delaney    32646  1697604        M   30   
667              Bruno Henrique    32115  1670000        F   30   
668                 Filipe Luís    32115  1670000        D   36   
669                       Jorge    32115  1670000        D   25   
670                Roger Guedes    32115  1670000        F   24   
671                     Vitinho    32115  1670000        F   27   
696             Matija Nastasic    31278  1626447        D   28   
697              Nahitan Nández    31278  1626447        M   26   
716             Andrej Kramaric    30105  1565456        F   30   
763                Jérémie Boga    29323  1524794        F   25   
832           Gonzalo Escalante    27368  1423141        M   28   
868          Kingsley Schindler    26586  1382480        F   28   
870                 Mario Götze    26586  1382480        F   29   
871                Miguel Borja    26538  1380000        F   28   
884                Fabian Schär    26432  1374453        D   30   
910             Ademola Lookman    25968  1350340        F   24   
930             Kevin Strootman    25413  1321488        M   31   
937                 Alan Pulido  25384,8  1320000        F   30   
948                       Pizzi    25022  1301158        F   32   
975                 Emre Kilinc    24436  1270662        F   27   
993       Julian Baumgartlinger    23849  1240166        M   34   
995               Stefan Lainer    23849  1240166        D   29   
1018              Mohamed Fares    23458  1219836        M   25   
1116               Maciej Rybus    21504  1118183        D   32   
1117          Manolo Gabbiadini    21504  1118183        F   30   
1118              Marco Benassi    21504  1118183        M   26   
1122          Riccardo Orsolini    21504  1118183        F   25   
1197                  Ché Adams    19940  1036868        F   25   
1225           Darlington Nagbe    19731  1026000        M   31   
1237           Cristian Ansaldi    19549  1016530        D   35   
1248          Jens Petter Hauge    19549  1016530        F   22   
1257      Lucas Martínez Quarta    19549  1016530        D   25   
1273            Roberto Pereyra    19549  1016530        F   31   
1301        Aleksandr Chernikov    18962   986034        M   22   
1324                Adam Masina    18549   964528        D   28   
1370          Cecilio Domínguez  17653,8   918000        F   27   
1396                  Moi Gómez    17594   914877        F   27   
1410             Connor Roberts    17389   904245        D   26   
1467 

In [33]:
data_c = data.query('country == "France"')
data_c

player   net_pw    net_py position  age country  \
0                  Kylian Mbappé   977432  50826480        F   23  France   
12                Raphaël Varane   225366  11719018        D   28  France   
13                 Karim Benzema   225200  11710421        F   34  France   
26             Antoine Griezmann   195486  10165296        F   30  France   
27              Presnel Kimpembe   195486  10165296        D   26  France   
31                    Paul Pogba   192210   9994924        M   28  France   
32                  N'Golo Kanté   191978   9982867        M   30  France   
33               Ousmane Dembélé   187667   9758684        F   24  France   
34               Lucas Hernández   186494   9697692        D   25  France   
38                Kingsley Coman   176133   9158932        F   25  France   
40               Anthony Martial   165778   8620471        F   25  France   
55                 Adrien Rabiot   136841   7115707        M   26  France   
62               Tanguy Ndombélé   132735   6902236        M   25  France   
63             Wissam Ben Yedder   132735   6902236        F   31  France   
82           Alexandre Lacazette   120798   6281490        F   30  France   
93               Clément Lenglet   112600   5855210        D   26  France   
96                   Lucas Digne   108509   5642490        D   28  France   
99               Dayot Upamecano   103608   5387607        D   23  France   
139                Ferland Mendy    87969   4574383        D   26  France   
162              Alphonse Areola    79527   4135415        K   28  France   
163                   Kurt Zouma    79527   4135415        D   27  France   
164                  Malang Sarr    79527   4135415        D   23  France   
177            Eduardo Camavinga    78195   4066118        M   19  France   
178               Theo Hernández    78195   4066118        D   24  France   
199             Corentin Tolisso    72525   3771325        M   27  France   
201             Francis Coquelin    72330   3761160        M   30  France   
202              Florian Thauvin    72308   3760000        F   29  France   
213               Olivier Giroud    68420   3557854        F   35  France   
226                  Hugo Lloris    66311   3448188        K   35  France   
238            Boubakary Soumaré    65616   3412019        M   22  France   
240                  Nabil Fekir    64706   3364713        F   28  France   
242                Dimitri Payet    64511   3354548        F   34  France   
248          André-Pierre Gignac    62885   3270000        F   36  France   
249              Odsonne Edouard    62833   3267339        F   24  France   
261                 Thomas Lemar    60992   3171572        F   26  France   
262                Steven Nzonzi    60601   3151242        M   32  France   
268               Benjamin Mendy    59587   3098547        D   27  France   
276              Jordan Veretout    58646   3049589        M   28  France   
277                Samuel Umtiti    58646   3049589        D   28  France   
297               Layvin Kurzawa    58059   3019093        D   29  France   
308                 Mike Maignan    54736   2846283        K   26  France   
309               Djibril Sidibé    54541   2836118        D   29  France   
316               Moussa Sissoko    53095   2760962        M   32  France   
334              Benjamin Pavard    51804   2693803        D   25  France   
358           Abdoulaye Doucouré    49849   2592170        M   29  France   
366             Tiemoué Bakayoko    48872   2541324        M   27  France   
401              Ibrahima Konaté    46372   2411321        D   22  France   
425                    Issa Diop    44285   2302811        D   25  France   
426               Moussa Dembélé    44180   2297357        F   25  France   
457                 Rémy Cabella    42421   2205869        F   31  France   
474                Adrien Trebel    41248   2144877        M   30  France   
485                Hou

In [34]:
data_pivot = pd.pivot_table(data_coutry_filtered,
               index='country',
               values="net_py",
               columns="rank"
               )

display(data_pivot.shape)

(84, 5)

In [35]:
zar = data.groupby(['country']).agg({'player':'count'})
zar.sort_values(by='player',ascending = False,inplace=True)
display(zar.shape)

(153, 1)

## 5.1. Национальный рейтинг по игрокам итоговая табица<a name="nazf2"></a>

In [36]:
rating_naz = data_pivot.merge(zp ,left_on='country' ,right_on = 'country')
display(rating_naz.shape)
rating_naz = rating_naz.sort_values(by='net_py',ascending = False)
rating_naz = rating_naz.fillna(0)
rating_naz 

(84, 6)

10.0       30.0       50.0     100.0  \
country                                                                       
France                           8620471.00 3171572.00 1844660.00 660744.00   
Argentina                        3252895.00 1423141.00 1016530.00 540000.00   
Brazil                           4808185.00 2390000.00 1670000.00 792893.00   
Portugal                         3181738.00 1301158.00  542547.00  38128.00   
Wales                             904245.00  192906.00       0.00      0.00   
Spain                            6363475.00 3293556.00 2155043.00 914877.00   
Belgium                          3098547.00  864050.00  355785.00      0.00   
Netherlands                      2592170.00 1321488.00  681075.00 132623.00   
Poland                           1118183.00  349641.00   60283.00      0.00   
England                          4574383.00 2033059.00 1350340.00 385811.00   
Germany                          5172283.00 2043224.00 1382480.00 498100.00   
Austria                          1240166.00  365951.00  111818.00      0.00   
Italy                            3029258.00 1118183.00  559091.00  12057.00   
Slovenia                          660000.00  135000.00       0.00      0.00   
Croatia                          1565456.00  518430.00  192906.00      0.00   
Uruguay                          1626447.00  620000.00  360000.00      0.00   
Gabon                              71157.00       0.00       0.00      0.00   
Morocco                           964528.00  355785.00   30496.00      0.00   
Denmark                          1697604.00  482264.00  156736.00      0.00   
Chile                             782728.00  160000.00       0.00      0.00   
Serbia                           1626447.00  367500.00   81322.00      0.00   
Sweden                            868075.00  254132.00   50826.00      0.00   
Egypt                             172810.00       0.00       0.00      0.00   
Ghana                             843720.00  233802.00   72340.00      0.00   
Costa Rica                        193141.00       0.00       0.00      0.00   
South Korea                       294794.00       0.00       0.00      0.00   
Senegal                          1839919.00  416777.00  142314.00      0.00   
Canada                            265245.00   57825.00       0.00      0.00   
United States                    1026000.00  508265.00  360000.00 115800.00   
Turkey                           1270662.00  559091.00  304959.00      0.00   
Bosnia-Herzegovina                626943.00  132149.00       0.00      0.00   
Colombia                         1380000.00  560000.00  340000.00  40661.00   
Cote d'Ivoire                    1524794.00  437108.00  111818.00      0.00   
Nigeria                           853885.00  376116.00   91488.00      0.00   
Mexico                           1320000.00  620000.00  390000.00      0.00   
Norway                           1016530.00  304959.00   24113.00      0.00   
Guinea                            193141.00       0.00       0.00      0.00   
Algeria                          1219836.00   60992.00       0.00      0.00   
Greece                            416777.00   84396.00       0.00      0.00   
Ukraine                           711571.00       0.00       0.00      0.00   
Scotland                         1036868.00  325528.00  120566.00      0.00   
Iceland                           264298.00       0.00       0.00      0.00   
Cameroon                          518430.00  101653.00       0.00      0.00   
Switzerland                      1374453.00  416777.00  203306.00      0.00   
Russia                           1768762.00  986034.00  650579.00  91488.00   
Slovakia                          487934.00       0.00       0.00      0.00   
Northern Ireland                  277302.00       0.00       0.00      0.00   
New Zealand                        60283.00       0.00       0.00      0.00   
Burkina Faso                       40661.00       0.00       0.00      0.00   
Democratic

## Вывод:
* По максимально зарплате в клубе лидеры Франция, Аргентина,Бразилия и Португалия, думаю можно без проблем называть этих игроков. Франция сильно в отрыве по игрокам из всех корзин. 

* Только у Бразили и Испании оказались 200е и 100е игроки у которых больши разброс по зп.

* Нули представлены здесь как замена пропущенным значениям, не все игроки  дотягивают до отметки 100е и 200е и ранк кидает их в другую корзину. Для того что бы проверить это оставил вычесления выше. 

* Получается, что несмотря на количество игроков зп могут быть сильно в разных диапозонах.

##  Найдем суммы зарплат 
(платежные ведомости) всех команд. 
* найти где граница между топ-клубами и просто сильными командами.

In [37]:
data_new = pd.read_csv('Capology_vedomosti.csv',sep = ",",low_memory=False)
data_new = data_new.replace("\'", '\"')

In [38]:
# с помощью лямб функции поменяем тип данных с object  на  float для того что бы произвести расчеты
#data_new['net_py'] = data_new['net_py'].apply(lambda x: x.replace('$', '').replace(',', '')).astype('float')
#data_new['net_pw'] = data_new['net_pw'].apply(lambda x: x.replace('$', '').replace(',', '')).astype('float')



In [39]:
# в расчет берем зп за год
total_salary = data_new.groupby(['club']).agg({'net_py':'sum'})
total_salary.sort_values(by='net_py',ascending = False,inplace=True)

In [40]:
total_salary.reset_index()

club     net_py
0                     PSG  240333009
1       Manchester United  163547819
2             Real Madrid  157389280
3           Bayern Munich  128184381
4                 Chelsea  115478143
5               Barcelona  114644205
6                Juventus  102262875
7         Manchester City   99485060
8               Liverpool   93089032
9             Inter Milan   85693444
10        Atletico Madrid   80651457
11                Arsenal   66353512
12                 Napoli   62602971
13              Tottenham   61627325
14                   Roma   57738879
15               AC Milan   56391980
16                Everton   53567484
17               West Ham   51385244
18             Villarreal   50567266
19              Leicester   48817187
20      Borussia Dortmund   48610448
21         Crystal Palace   48383147
22                 Monaco   45260982
23                Sevilla   44783210
24                  Lazio   42973788
25               Besiktas   40295234
26            Aston Villa   37845676
27              Newcastle   37743192
28             Fiorentina   36767875
29          Wolverhampton   36344627
30            Southampton   34903869
31                   Lyon   31842789
32                  Zenit   31832623
33          Athletic Club   30668694
34               Valencia   29032083
35              Marseille   28981257
36               Brighton   28748969
37            Galatasaray   28666133
38              Wolfsburg   28579729
39             Fenerbahce   28335764
40                Leipzig   28320515
41               Flamengo   27125000
42                 Torino   27029521
43                 Fulham   26548642
44       Bayer Leverkusen   26516173
45               Atalanta   25692785
46                Watford   25686591
47         Spartak Moscow   24869396
48                Burnley   24794402
49        Monchengladbach   24147661
50          Real Sociedad   23502163
51             Real Betis   23375098
52            Trabzonspor   22790597
53            Bournemouth   22371026
54              Krasnodar   22129851
55       Atletico Mineiro   21300000
56             Hoffenheim   21214973
57                Benfica   21108238
58             Schalke 04   20544065
59               Cagliari   20411912
60                Norwich   20224950
61                  Genoa   20000220
62              Lokomotiv   19944311
63                 Empoli   19542785
64              Palmeiras   19190000
65    Eintracht Frankfurt   19141255
66                Bologna   19100592
67                   CSKA   18216210
68                   Ajax   18028153
69                  Parma   17987490
70          Hertha Berlin   17962076
71             Celta Vigo   17824847
72          Dynamo Moscow   17707949
73              Sampdoria   17591044
74        Adana Demirspor   16813399
75                   UANL   16650000
76            Corinthians   16475000
77            River Plate   16360000
78             Basaksehir   16259391
79                Rangers   15993082
80                  Leeds   15794148
81                   Nice   15659639
82               Sassuolo   15547823
83                 Getafe   15527488
84                   Koln   14836251
85                  Porto   14760008
86                  Lille   14744759
87                Venezia   14521126
88         Karagumrukspor   14470302
89          West Bromwich   14419696
90               Espanyol   14302571
91               Hannover   14185670
92              Sao Paulo   13915000
93                America   13810000
94           Gaziantep BB   13763810
95               Bordeaux   13479182
96           Boca Juniors   13460000
97              Monterrey   13415000
98          Internacional   13295000
99                  Monza   13174224
100             Brentford   13075381
101            Stoke City   12990986
102               Udinese   12894681
103      Sheffield United   12876450
104     Nottingham Forest   12810141
105               Reading   12798082
106              Augsburg   12721868
107                Rennes

## Вывод:
По таблице можно сказать что суммы от 48569786 млн евро зарплатной ведомости - это топ клубы, после этой отметки категория сильные клубы. ПСЖ вне конкуренции - сильный отрыв по зп. Остальные топ 4 имеют примерно одинаковую зп ведомость, между остальными топ клубами разница существенная.

Наш топ 5 довольно очевидный:
* PSG	240439745
* Manchester United	163547819
* Real Madrid	157389280
* Bayern Munich	128184381
* Chelsea 116225653

## Посмотрим на таблицу более подробно
Нужно найти:

* где будут российские клубы, Аякс, Спортинг, Зальцбург, Шахтер 

* Где будет первый клуб МЛС, первый мексиканский и т.д.

In [41]:
#найдем сумму платежной ведомости клуба
total_sum_test = data_new.groupby(['club','league']).agg({'net_py':'sum'})
total_sum_test.sort_values(by='net_py',ascending = False,inplace=True)
total_sum_test.head(3)

net_py
club              league                   
PSG               Ligue 1         240333009
Manchester United Premier League  163547819
Real Madrid       La Liga         157389280

In [42]:
#найдем максимальную зп
total_sum_2 = data_new.groupby(['club','league']).agg({'net_py':'max'})
total_sum_2.sort_values(by='net_py',ascending = False,inplace=True)

In [43]:
test_two = total_sum_test.merge(total_sum_2,left_on=['club','league'],right_on=['club','league'],how='right')
display(test_two.shape)

(488, 2)

In [44]:
total = test_two.rename({'net_py_x':'net_py','net_py_y':'net_pw'}, axis=1)

In [45]:
display(total.shape)
display(total.head(4))



(488, 2)

net_py    net_pw
club              league                             
PSG               Ligue 1         240333009  50826480
Manchester United Premier League  163547819  30141507
Real Madrid       La Liga         157389280  16589763
Barcelona         La Liga         114644205  14231414

In [46]:
#сохраняем таблицу с чтобы отправить себе и потом загрузим снова с индексами
total.to_csv(r'total_ver2.csv')

In [47]:
#проставим индексы места в рейтинге
dt = pd.read_csv('total_ver2.csv',sep = ",",index_col = 0,low_memory=False)
dt = dt.replace("\'", '\"')

In [48]:
dt.groupby('club')
dt = dt.sort_values(by='net_py',ascending = False)
dt.reset_index()

club               league     net_py    net_pw
0                     PSG              Ligue 1  240333009  50826480
1       Manchester United       Premier League  163547819  30141507
2             Real Madrid              La Liga  157389280  16589763
3           Bayern Munich         1.Bundesliga  128184381  12391496
4                 Chelsea       Premier League  115478143  11212641
5               Barcelona              La Liga  114644205  14231414
6                Juventus              Serie A  102262875   8132237
7         Manchester City       Premier League   99485060  13792754
8               Liverpool       Premier League   93089032   7583603
9             Inter Milan              Serie A   85693444   7115707
10        Atletico Madrid              La Liga   80651457  10165296
11                Arsenal       Premier League   66353512   6896377
12                 Napoli              Serie A   62602971   6099178
13              Tottenham       Premier League   61627325   6896377
14                   Roma              Serie A   57738879   4574383
15               AC Milan              Serie A   56391980   7115707
16                Everton       Premier League   53567484   4135415
17               West Ham       Premier League   51385244   4135415
18             Villarreal              La Liga   50567266   5285954
19              Leicester       Premier League   48817187   4822641
20      Borussia Dortmund         1.Bundesliga   48610448   5926368
21         Crystal Palace       Premier League   48383147   4485056
22                 Monaco              Ligue 1   45260982   6902236
23                Sevilla              La Liga   44783210   6444798
24                  Lazio              Serie A   42973788   4066118
25               Besiktas            Super Lig   40295234   4726863
26            Aston Villa       Premier League   37845676   4135415
27              Newcastle       Premier League   37743192   2760962
28             Fiorentina              Serie A   36767875   3557854
29          Wolverhampton       Premier League   36344627   4509170
30            Southampton       Premier League   34903869   2592170
31                   Lyon              Ligue 1   31842789   3451118
32                  Zenit         Premier Liga   31832623   7075046
33          Athletic Club              La Liga   30668694   4055953
34               Valencia              La Liga   29032083   2978432
35              Marseille              Ligue 1   28981257   3354548
36               Brighton       Premier League   28748969   3122660
37            Galatasaray            Super Lig   28666133   3252895
38              Wolfsburg         1.Bundesliga   28579729   2968266
39             Fenerbahce            Super Lig   28335764   3374878
40                Leipzig         1.Bundesliga   28320515   4066118
41               Flamengo           Brasileiro   27125000   2680000
42                 Torino              Serie A   27029521   2033059
43                 Fulham         Championship   26548642   2242528
44       Bayer Leverkusen         1.Bundesliga   26516173   1728100
45               Atalanta              Serie A   25692785   2033059
46                Watford       Premier League   25686591   2760962
47         Spartak Moscow         Premier Liga   24869396   3049589
48                Burnley       Premier League   24794402   1892887
49        Monchengladbach         1.Bundesliga   24147661   2266861
50          Real Sociedad              La Liga   23502163   2124547
51             Real Betis              La Liga   23375098   3364713
52            Trabzonspor            Super Lig   22790597   2541324
53            Bournemouth         Championship   22371026   1892887
54              Krasnodar         Premier Liga   22129851   3039424
55       Atletico Mineiro           Brasileiro   21300000   2680000
56             Hoffenheim         1.Bundesliga   21214973   2368514
57                Benfica        Primeira Liga   21108238   2134712
58       

* первый клуб не из топ 5 это Бешикташ 26 место, следом Зенит 33 место.
* Фламенго 42 место * Спартак  - 48 место * Краснодар - 68 место * Аякс - 63 место  * Локомотив - 66 место * ЦСКА- 72 место * Динамо - 74 место * Рейнджерс - 77 место * Мексиканский UANL - 80 место * Орландо - 91 место *  Арсенал Тула - 114 место * Спортинг - 117 * Ростов - 122 * Сочи - 123 * Рубин - 135 * Ахмат - 143 * * Базель - 160 * Уфа 182 * Нижний - 206 * Урал - 217  * Крылья - 275 

## 6.Рейтинг клубов по ЗП ведомости<a name="view2"></a>

Задача: Собрать таблицу с колонками

- место в рейтинге - клуб - место в лиге по зп - размер ведомости net - максимальная зп - Перцентиль 90	- Перцентиль 80 - Перцентиль 60	- Перцентиль 40

Шаги:
1.Собрать зп ведомость по лигам и распределить клубы 
2.Собрать размезмер зп ведомости (сделано)
3.Найти максимальную зп
4.Найти перцентили и добавить в таблицу

In [49]:
percentil_40 = data_new.groupby('club')['net_py'].quantile(0.4)
percentil_60 = data_new.groupby('club')['net_py'].quantile(0.6)
percentil_80 = data_new.groupby('club')['net_py'].quantile(0.8)
percentil_90 = data_new.groupby('club')['net_py'].quantile(0.9)

In [50]:
#соберем перцентили
data_perc = pd.DataFrame()
data_perc['percentil40'] = percentil_40
data_perc['percentil60'] = percentil_60
data_perc['percentil80'] = percentil_80 
data_perc['percentil90'] = percentil_90
display(data_perc.shape)

(488, 4)

In [51]:
#максимальная зарпатная ведомость по команде
maxzp = data_new.groupby(['club','league']).agg({'net_pw':'max'})
maxzp.sort_values(by='net_pw',ascending = False,inplace=True)
display(maxzp.shape)
maxzp 

(488, 1)

net_pw
club                 league                     
Cardiff              Championship           9970
Karagumrukspor       Super Lig              9970
Bournemouth          Championship           9970
Granada              La Liga                9970
Celtic               Premiership            9970
Hoffenheim           1.Bundesliga           9970
Newcastle            Premier League         9970
Nottingham Forest    Championship           9970
Huddersfield         Championship           9970
CSKA                 Premier Liga           9970
Reading              Championship           9970
Sevilla              La Liga                9970
Rangers              Premiership            9970
Koln                 1.Bundesliga           9970
Preston              Championship           9970
Lecce                Serie B                9970
Bristol              Championship           9970
Brighton             Premier League         9970
PSV                  Eredivisie             9970
Brest                Ligue 1                9970
Leeds                Premier League         9970
Sheffield United     Championship           9970
Brentford            Premier League         9970
Twente               Eredivisie             9970
Yeni Malatyaspor     Super Lig              9970
Wolverhampton        Premier League         9970
Derby County         Championship           9970
Watford              Premier League         9970
Everton              Premier League         9970
Ufa                  Premier Liga           9970
Ajax                 Eredivisie             9970
Basel                Super League           9970
Tottenham            Premier League         9970
Sheffield Wednesday  League One             9970
Mainz                1.Bundesliga           9970
Swansea              Championship           9970
Strasbourg           Ligue 1                9970
Stoke City           Championship           9970
Birmingham           Championship           9970
Blackburn            Championship           9970
Fenerbahce           Super Lig              9970
Liverpool            Premier League         9970
Nurnberg             2.Bundesliga           9970
Manchester City      Premier League        99467
Manchester United    Premier League        99467
Chelsea              Premier League        99467
San Jose Earthquakes MLS                  9922,8
Montreal Impact      MLS                   986,4
LA Galaxy            MLS                   985,8
LA FC                MLS                   985,8
Houston Dynamo       MLS                   985,8
Chicago Fire         MLS                   985,8
FC Dallas            MLS                   985,8
FC Cincinnati        MLS                   985,8
Colorado Rapids      MLS                   985,8
Columbus Crew        MLS                   985,8
DC United            MLS                   985,8
Real Salt Lake       MLS                   985,8
Portland Timbers     MLS                   985,8
NY Red Bulls         MLS                   985,8
NYC FC               MLS                   985,8
Toronto FC           MLS                   985,8
Sporting KC          MLS                   985,8
Orlando City         MLS                   985,8
Nashville SC         MLS                   985,8
Austin               MLS                     981
Independiente        Argentinian Primera    9808
Racing Club          Argentinian Primera    9808
Internacional        Brasileiro             9808
Atletico Nacional    Colombia               9808
Lanús                Argentinian Primera    9808
Guadalajara          Liga MX                9808
Leon                 Liga MX                9808
Gremio               Brasileiro             9808
Sport Club do Recife Brasileiro             9808
Boca Juniors         Argentinian Primera    9808
Estudiantes          Argentinian Primera    9808
Monterrey            Liga MX                9808
Millonarios          Colombia               9808
Minnesota United     MLS                  9807,6
Inter Miami          MLS

In [52]:
#соеденим таблицы
test_2 = maxzp.merge(dt,left_on='club',right_on='club',how='right')
display(test_2.shape)
test_2 

(488, 4)

net_pw_x               league     net_py  net_pw_y
club                                                                   
PSG                    977432              Ligue 1  240333009  50826480
Manchester United       99467       Premier League  163547819  30141507
Real Madrid             92465              La Liga  157389280  16589763
Bayern Munich             977         1.Bundesliga  128184381  12391496
Chelsea                 99467       Premier League  115478143  11212641
Barcelona               97743              La Liga  114644205  14231414
Juventus                97743              Serie A  102262875   8132237
Manchester City         99467       Premier League   99485060  13792754
Liverpool                9970       Premier League   93089032   7583603
Inter Milan             97743              Serie A   85693444   7115707
Atletico Madrid          7819              La Liga   80651457  10165296
Arsenal                 92743       Premier League   66353512   6896377
Napoli                  89924              Serie A   62602971   6099178
Tottenham                9970       Premier League   61627325   6896377
Roma                     9774              Serie A   57738879   4574383
AC Milan                97743              Serie A   56391980   7115707
Everton                  9970       Premier League   53567484   4135415
West Ham                79527       Premier League   51385244   4135415
Villarreal              72330              La Liga   50567266   5285954
Leicester               92743       Premier League   48817187   4822641
Borussia Dortmund        9188         1.Bundesliga   48610448   5926368
Crystal Palace          86251       Premier League   48383147   4485056
Monaco                   9774              Ligue 1   45260982   6902236
Sevilla                  9970              La Liga   44783210   6444798
Lazio                    9774              Serie A   42973788   4066118
Besiktas                 9774            Super Lig   40295234   4726863
Aston Villa               927       Premier League   37845676   4135415
Newcastle                9970       Premier League   37743192   2760962
Fiorentina               7819              Serie A   36767875   3557854
Wolverhampton            9970       Premier League   36344627   4509170
Southampton              8347       Premier League   34903869   2592170
Lyon                     9774              Ligue 1   31842789   3451118
Zenit                     977         Premier Liga   31832623   7075046
Athletic Club           77999              La Liga   30668694   4055953
Valencia                 8210              La Liga   29032083   2978432
Marseille                9579              Ligue 1   28981257   3354548
Brighton                 9970       Premier League   28748969   3122660
Galatasaray              8797            Super Lig   28666133   3252895
Wolfsburg                9383         1.Bundesliga   28579729   2968266
Fenerbahce               9970            Super Lig   28335764   3374878
Leipzig                 78195         1.Bundesliga   28320515   4066118
Flamengo                 8462           Brasileiro   27125000   2680000
Torino                   7819              Serie A   27029521   2033059
Fulham                   9042         Championship   26548642   2242528
Bayer Leverkusen         9774         1.Bundesliga   26516173   1728100
Atalanta                 9774              Serie A   25692785   2033059
Watford                  9970       Premier League   25686591   2760962
Spartak Moscow           9188         Premier Liga   24869396   3049589
Burnley                  7883       Premier League   24794402   1892887
Monchengladbach           586         1.Bundesliga   24147661   2266861
Real Sociedad            9774              La Liga   23502163   2124547
Real Betis               9774              La Liga   23375098   3364713
Trabzonspor               977            Super Lig   22790597   2541324
Bournemouth              9970         Championship   22371026

In [53]:
#соеденим таблицы
df_merge = test_2.merge(data_perc,left_on='club' ,right_on = 'club' )
#df_merge.sort_values(by=['net_py'],ascending=False).reset_index()
df_merge.groupby('club')
display(df_merge.shape)
df_merge 

(488, 8)

net_pw_x               league     net_py  net_pw_y  \
club                                                                      
PSG                    977432              Ligue 1  240333009  50826480   
Manchester United       99467       Premier League  163547819  30141507   
Real Madrid             92465              La Liga  157389280  16589763   
Bayern Munich             977         1.Bundesliga  128184381  12391496   
Chelsea                 99467       Premier League  115478143  11212641   
Barcelona               97743              La Liga  114644205  14231414   
Juventus                97743              Serie A  102262875   8132237   
Manchester City         99467       Premier League   99485060  13792754   
Liverpool                9970       Premier League   93089032   7583603   
Inter Milan             97743              Serie A   85693444   7115707   
Atletico Madrid          7819              La Liga   80651457  10165296   
Arsenal                 92743       Premier League   66353512   6896377   
Napoli                  89924              Serie A   62602971   6099178   
Tottenham                9970       Premier League   61627325   6896377   
Roma                     9774              Serie A   57738879   4574383   
AC Milan                97743              Serie A   56391980   7115707   
Everton                  9970       Premier League   53567484   4135415   
West Ham                79527       Premier League   51385244   4135415   
Villarreal              72330              La Liga   50567266   5285954   
Leicester               92743       Premier League   48817187   4822641   
Borussia Dortmund        9188         1.Bundesliga   48610448   5926368   
Crystal Palace          86251       Premier League   48383147   4485056   
Monaco                   9774              Ligue 1   45260982   6902236   
Sevilla                  9970              La Liga   44783210   6444798   
Lazio                    9774              Serie A   42973788   4066118   
Besiktas                 9774            Super Lig   40295234   4726863   
Aston Villa               927       Premier League   37845676   4135415   
Newcastle                9970       Premier League   37743192   2760962   
Fiorentina               7819              Serie A   36767875   3557854   
Wolverhampton            9970       Premier League   36344627   4509170   
Southampton              8347       Premier League   34903869   2592170   
Lyon                     9774              Ligue 1   31842789   3451118   
Zenit                     977         Premier Liga   31832623   7075046   
Athletic Club           77999              La Liga   30668694   4055953   
Valencia                 8210              La Liga   29032083   2978432   
Marseille                9579              Ligue 1   28981257   3354548   
Brighton                 9970       Premier League   28748969   3122660   
Galatasaray              8797            Super Lig   28666133   3252895   
Wolfsburg                9383         1.Bundesliga   28579729   2968266   
Fenerbahce               9970            Super Lig   28335764   3374878   
Leipzig                 78195         1.Bundesliga   28320515   4066118   
Flamengo                 8462           Brasileiro   27125000   2680000   
Torino                   7819              Serie A   27029521   2033059   
Fulham                   9042         Championship   26548642   2242528   
Bayer Leverkusen         9774         1.Bundesliga   26516173   1728100   
Atalanta                 9774              Serie A   25692785   2033059   
Watford                  9970       Premier League   25686591   2760962   
Spartak Moscow           9188         Premier Liga   24869396   3049589   
Burnley                  7883       Premier League   24794402   1892887   
Monchengladbach           586         1.Bundesliga   24147661   2266861   
Real Sociedad            9774              La Liga   23502163   2124547   
Real Betis               9774              La 

In [54]:
df_merge = df_merge.drop(['net_pw_x'],axis=1)

In [55]:
final = df_merge
#.rename({'club':'клуб','net_pw_y':'макс зп','ligue':'лига','net_py':'размер ведомости','percentil40':'перцентиль40','percentil60':'перцентиль60','percentil80':'перцентиль80','percentil90':'перцентиль90'}, axis=1)
display(final.shape)
final.head(2)

(488, 7)

league     net_py  net_pw_y  percentil40  \
club                                                                  
PSG                       Ligue 1  240333009  50826480    527578.60   
Manchester United  Premier League  163547819  30141507   2963513.00   

                   percentil60  percentil80  percentil90  
club                                                      
PSG                 4159639.00   8050914.00  10673560.80  
Manchester United   4205343.00   8340758.20  11753982.20

In [56]:
#создадим фильтр лист для ранк
filter_list_rank=[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30]

In [57]:
dt_ranked = final.copy()
dt_ranked['rank'] = final.groupby('league')['net_py'].rank(method='dense',ascending=False)
dt_rank_filtered = dt_ranked.query("rank == @filter_list_rank")
#dt_rank_filtered.sort_values(by = ['net_py'], ascending = False)
dt_rank_filtered.head(1)

league     net_py  net_pw_y  percentil40  percentil60  percentil80  \
club                                                                        
PSG   Ligue 1  240333009  50826480    527578.60   4159639.00   8050914.00   

      percentil90  rank  
club                     
PSG   10673560.80  1.00

### 6.1 Итог таблица перцентили<a name="final1"></a>

In [58]:
dt_rank_filtered = dt_rank_filtered.rename({'club':'клуб','ligue':'лига','net_pw_y':'макс зп','net_py':'размер ведомости','percentil40':'перцентиль40','percentil60':'перцентиль60','percentil80':'перцентиль80','percentil90':'перцентиль90'}, axis=1)
dt_rank_filtered = dt_rank_filtered.sort_values(by = 'размер ведомости', ascending = False)
dt_rank_filtered

league  размер ведомости   макс зп  \
club                                                                    
PSG                               Ligue 1         240333009  50826480   
Manchester United          Premier League         163547819  30141507   
Real Madrid                       La Liga         157389280  16589763   
Bayern Munich                1.Bundesliga         128184381  12391496   
Chelsea                    Premier League         115478143  11212641   
Barcelona                         La Liga         114644205  14231414   
Juventus                          Serie A         102262875   8132237   
Manchester City            Premier League          99485060  13792754   
Liverpool                  Premier League          93089032   7583603   
Inter Milan                       Serie A          85693444   7115707   
Atletico Madrid                   La Liga          80651457  10165296   
Arsenal                    Premier League          66353512   6896377   
Napoli                            Serie A          62602971   6099178   
Tottenham                  Premier League          61627325   6896377   
Roma                              Serie A          57738879   4574383   
AC Milan                          Serie A          56391980   7115707   
Everton                    Premier League          53567484   4135415   
West Ham                   Premier League          51385244   4135415   
Villarreal                        La Liga          50567266   5285954   
Leicester                  Premier League          48817187   4822641   
Borussia Dortmund            1.Bundesliga          48610448   5926368   
Crystal Palace             Premier League          48383147   4485056   
Monaco                            Ligue 1          45260982   6902236   
Sevilla                           La Liga          44783210   6444798   
Lazio                             Serie A          42973788   4066118   
Besiktas                        Super Lig          40295234   4726863   
Aston Villa                Premier League          37845676   4135415   
Newcastle                  Premier League          37743192   2760962   
Fiorentina                        Serie A          36767875   3557854   
Wolverhampton              Premier League          36344627   4509170   
Southampton                Premier League          34903869   2592170   
Lyon                              Ligue 1          31842789   3451118   
Zenit                        Premier Liga          31832623   7075046   
Athletic Club                     La Liga          30668694   4055953   
Valencia                          La Liga          29032083   2978432   
Marseille                         Ligue 1          28981257   3354548   
Brighton                   Premier League          28748969   3122660   
Galatasaray                     Super Lig          28666133   3252895   
Wolfsburg                    1.Bundesliga          28579729   2968266   
Fenerbahce                      Super Lig          28335764   3374878   
Leipzig                      1.Bundesliga          28320515   4066118   
Flamengo                       Brasileiro          27125000   2680000   
Torino                            Serie A          27029521   2033059   
Fulham                       Championship          26548642   2242528   
Bayer Leverkusen             1.Bundesliga          26516173   1728100   
Atalanta                          Serie A          25692785   2033059   
Watford                    Premier League          25686591   2760962   
Spartak Moscow               Premier Liga          24869396   3049589   
Burnley                    Premier League          24794402   1892887   
Monchengladbach              1.Bundesliga          24147661   2266861   
Real Sociedad                     La Liga          23502163   2124547   
Real Betis                        La Liga          23375098   3364713   
Trabzonspor                     Super Lig          22790597   2541324   
Bournemouth         

### Теперь соберем таблицу по игрокам

In [59]:
# отсортируем лиги по зп
total_ligue = dt.groupby(['league']).agg({'net_py':'sum'})
total_ligue.sort_values(by='net_py',ascending = False,inplace=True)
total_ligue.head(2)

net_py
league                    
Premier League  1076895758
La Liga          670741783

In [60]:
filter_list=[1,3,7,12,18]

#df_new.query("rank == @filter_list")

In [61]:
data_ranked = data_new.copy()
data_ranked['rank'] = data_new.groupby('club')['net_py'].rank(method='dense',ascending=False)
data_rank_filtered = data_ranked.query("rank == @filter_list")
display(data_rank_filtered.shape)
data_rank_filtered.sort_values(by = ['club','net_py'], ascending = False)
data_rank_filtered.head(1)

(3268, 9)

player  net_pw    net_py posotion  age country club   league  rank
0  Kylian Mbappé  977432  50826480        F   23  France  PSG  Ligue 1  1.00

In [62]:
statistic_pivot = pd.pivot_table(data_rank_filtered  ,
               index=['league','club'],
               values="net_py",
               columns="rank"
               )

display(statistic_pivot.shape)
statistic_pivot.head(2)

(488, 5)

rank                                1.00      3.00      7.00      12.00  \
league       club                                                         
1.Bundesliga Arminia Bielefeld  650579.00 487934.00 203306.00 111818.00   
             Augsburg          1128348.00 813224.00 487934.00 315124.00   

rank                               18.00  
league       club                         
1.Bundesliga Arminia Bielefeld       NaN  
             Augsburg          223637.00

In [63]:
final_player = statistic_pivot.merge(dt_rank_filtered,left_on=['league','club'],right_on=['league','club'],how='left').reset_index()
final_player = final_player[['league','club', 'rank','размер ведомости' ,1.0, 3.0, 7.0, 12.0, 18.0]]
#final_player = final_player.sort_values(by = ['net_py'], ascending = False)
final_player = final_player.fillna(0)
final_player 

league                  club  rank  размер ведомости  \
0           1.Bundesliga     Arminia Bielefeld 18.00           4930166   
1           1.Bundesliga              Augsburg 11.00          12721868   
2           1.Bundesliga      Bayer Leverkusen  5.00          26516173   
3           1.Bundesliga         Bayern Munich  1.00         128184381   
4           1.Bundesliga                Bochum 15.00           6937815   
5           1.Bundesliga     Borussia Dortmund  2.00          48610448   
6           1.Bundesliga   Eintracht Frankfurt  8.00          19141255   
7           1.Bundesliga              Freiburg 16.00           6236411   
8           1.Bundesliga                 Furth 17.00           6150005   
9           1.Bundesliga         Hertha Berlin  9.00          17962076   
10          1.Bundesliga            Hoffenheim  7.00          21214973   
11          1.Bundesliga                  Koln 10.00          14836251   
12          1.Bundesliga               Leipzig  4.00          28320515   
13          1.Bundesliga                 Mainz 13.00           8848889   
14          1.Bundesliga       Monchengladbach  6.00          24147661   
15          1.Bundesliga             Stuttgart 14.00           8020420   
16          1.Bundesliga          Union Berlin 12.00          10287280   
17          1.Bundesliga             Wolfsburg  3.00          28579729   
18          2.Bundesliga             Darmstadt 11.00           5179217   
19          2.Bundesliga            Dusseldorf  5.00          11476618   
20          2.Bundesliga        Dynamo Dresden 16.00           3075001   
21          2.Bundesliga        Erzgebirge Aue 13.00           4244009   
22          2.Bundesliga               Hamburg  4.00          11832402   
23          2.Bundesliga              Hannover  2.00          14185670   
24          2.Bundesliga         Hansa Rostock 17.00           2897112   
25          2.Bundesliga            Heidenheim  8.00           5590912   
26          2.Bundesliga         Holstein Kiel 10.00           5433352   
27          2.Bundesliga            Ingolstadt 14.00           4132190   
28          2.Bundesliga                  Jahn 15.00           3639174   
29          2.Bundesliga            Karlsruher 18.00           2114377   
30          2.Bundesliga              Nurnberg  6.00           9280913   
31          2.Bundesliga             Paderborn  9.00           5509585   
32          2.Bundesliga            Sandhausen 12.00           4706530   
33          2.Bundesliga            Schalke 04  1.00          20544065   
34          2.Bundesliga              St Pauli  7.00           8101740   
35          2.Bundesliga         Werder Bremen  3.00          12472818   
36   Argentinian Primera              Aldosivi 28.00           2035000   
37   Argentinian Primera    Argentinos Juniors 15.00           3740000   
38   Argentinian Primera   Arsenal Fútbol Club 13.00           3870000   
39   Argentinian Primera              Banfield 14.00           3855000   
40   Argentinian Primera              Barracas 22.00           2950000   
41   Argentinian Primera          Boca Juniors  2.00          13460000   
42   Argentinian Primera       Central Córdoba 18.00           3350000   
43   Argentinian Primera                 Colón  9.00           5495000   
44   Argentinian Primera    Defensa y Justicia 23.00           2900000   
45   Argentinian Primera           Estudiantes  8.00           5655000   
46   Argentinian Primera     Gimnasia La Plata 12.00           3935000   
47   Argentinian Primera            Godoy Cruz 20.00           3055000   
48   Argentinian Primera               Huracán 27.00           2565000   
49   Argentinian Primera         Independiente  6.00           8410000   
50   Argentinian Primera                 Lanús  7.00           6195000   
51   Argentinian Primera     Newell's Old Boys 21.00           3050000   
52   Argentinian Primera             Patronato 25.00           2590000   
53   Argentinian Prime

## 6.2.Итог таблица по игрокам<a name="final2"></a>

In [64]:
final_player_2 = final_player.rename({'league':'лига','club':'клуб','rank':'ранг','net_py':'размер ведомости',1.0:'зп 1 игрока',3.0:'зп 3 игрока',7.0:'зп 7 игрока',12.0:'зп 12 игрока',18.0:'зп 18 игрока'}, axis=1)
final_player_2 = final_player.sort_values(by = 'размер ведомости', ascending = False)
final_player_2.reset_index()

index               league                  club  rank  размер ведомости  \
0      299              Ligue 1                   PSG  1.00         240333009   
1      364       Premier League     Manchester United  1.00         163547819   
2      192              La Liga           Real Madrid  1.00         157389280   
3        3         1.Bundesliga         Bayern Munich  1.00         128184381   
4      357       Premier League               Chelsea  2.00         115478143   
5      180              La Liga             Barcelona  2.00         114644205   
6      427              Serie A              Juventus  1.00         102262875   
7      363       Premier League       Manchester City  3.00          99485060   
8      362       Premier League             Liverpool  4.00          93089032   
9      426              Serie A           Inter Milan  2.00          85693444   
10     179              La Liga       Atletico Madrid  3.00          80651457   
11     352       Premier League               Arsenal  5.00          66353512   
12     429              Serie A                Napoli  3.00          62602971   
13     368       Premier League             Tottenham  6.00          61627325   
14     430              Serie A                  Roma  4.00          57738879   
15     418              Serie A              AC Milan  5.00          56391980   
16     359       Premier League               Everton  7.00          53567484   
17     370       Premier League              West Ham  8.00          51385244   
18     196              La Liga            Villarreal  4.00          50567266   
19     361       Premier League             Leicester  9.00          48817187   
20       5         1.Bundesliga     Borussia Dortmund  2.00          48610448   
21     358       Premier League        Crystal Palace 10.00          48383147   
22     295              Ligue 1                Monaco  2.00          45260982   
23     194              La Liga               Sevilla  5.00          44783210   
24     428              Serie A                 Lazio  6.00          42973788   
25     473            Super Lig              Besiktas  1.00          40295234   
26     353       Premier League           Aston Villa 11.00          37845676   
27     365       Premier League             Newcastle 12.00          37743192   
28     423              Serie A            Fiorentina  7.00          36767875   
29     371       Premier League         Wolverhampton 13.00          36344627   
30     367       Premier League           Southampton 14.00          34903869   
31     292              Ligue 1                  Lyon  3.00          31842789   
32     387         Premier Liga                 Zenit  1.00          31832623   
33     178              La Liga         Athletic Club  6.00          30668694   
34     195              La Liga              Valencia  7.00          29032083   
35     293              Ligue 1             Marseille  4.00          28981257   
36     355       Premier League              Brighton 15.00          28748969   
37     476            Super Lig           Galatasaray  2.00          28666133   
38      17         1.Bundesliga             Wolfsburg  3.00          28579729   
39     475            Super Lig            Fenerbahce  3.00          28335764   
40      12         1.Bundesliga               Leipzig  4.00          28320515   
41      72           Brasileiro              Flamengo  1.00          27125000   
42     435              Serie A                Torino  8.00          27029521   
43     105         Championship                Fulham  1.00          26548642   
44       2         1.Bundesliga      Bayer Leverkusen  5.00          26516173   
45     419              Serie A              Atalanta  9.00          25692785   
46     369       Premier League               Watford 16.00          25686591   
47     384         Premier Liga        Spartak Moscow  2.00          24869396   
48     356       Premier League     

In [83]:
qq = data_new.query('club =="PSG"')
qq.sort_values(by = 'rank',ascending = False)

#а data_new?

KeyError: 'rank'

In [65]:
club = data_new.groupby(['club','league','player']).agg({'net_py':'sum'})
club.sort_values(by='net_py',ascending = False,inplace=True)
club.head(160)

net_py
club              league         player                             
PSG               Ligue 1        Kylian Mbappé              50826480
                                 Lionel Messi               35578536
                                 Neymar                     31512418
Manchester United Premier League Cristiano Ronaldo          30141507
Real Madrid       La Liga        Gareth Bale                16589763
PSG               Ligue 1        Sergio Ramos               15247944
Real Madrid       La Liga        Eden Hazard                15247944
Barcelona         La Liga        Frenkie de Jong            14231414
Manchester City   Premier League Kevin De Bruyne            13792754
Manchester United Premier League David de Gea               12936735
Bayern Munich     1.Bundesliga   Robert Lewandowski         12391496
Manchester United Premier League Jadon Sancho               12068660
                                 Raphaël Varane             11719018
Real Madrid       La Liga        Karim Benzema              11710421
Bayern Munich     1.Bundesliga   Manuel Neuer               11313974
Chelsea           Premier League Romelu Lukaku              11212641
Bayern Munich     1.Bundesliga   Thomas Müller              11049677
Real Madrid       La Liga        David Alaba                10978520
Bayern Munich     1.Bundesliga   Leroy Sané                 10775214
Barcelona         La Liga        Sergio Busquets            10734553
Real Madrid       La Liga        Marcelo                    10704057
                                 Toni Kroos                 10704057
Bayern Munich     1.Bundesliga   Joshua Kimmich             10510916
Manchester City   Premier League Jack Grealish              10344565
                                 Raheem Sterling            10344565
PSG               Ligue 1        Presnel Kimpembe           10165296
Atletico Madrid   La Liga        Antoine Griezmann          10165296
PSG               Ligue 1        Gianluigi Donnarumma       10165296
Barcelona         La Liga        Jordi Alba                 10165296
Atletico Madrid   La Liga        Jan Oblak                  10144965
Manchester United Premier League Paul Pogba                  9994924
Chelsea           Premier League N'Golo Kanté                9982867
Barcelona         La Liga        Ousmane Dembélé             9758684
Bayern Munich     1.Bundesliga   Leon Goretzka               9697692
                                 Lucas Hernández             9697692
Real Madrid       La Liga        Luka Modric                 9606205
Chelsea           Premier League Timo Werner                 9380037
Bayern Munich     1.Bundesliga   Kingsley Coman              9158932
Manchester United Premier League Edinson Cavani              8620471
Manchester City   Premier League John Stones                 8620471
Manchester United Premier League Bruno Fernandes             8270830
Juventus          Serie A        Matthijs de Ligt            8132237
PSG               Ligue 1        Achraf Hakimi               8132237
                                 Marco Verratti              8050914
                                 Marquinhos                  8050914
Atletico Madrid   La Liga        Koke                        7806947
                                 Luis Suárez                 7623972
Liverpool         Premier League Virgil van Dijk             7583603
Juventus          Serie A        Paulo Dybala                7420666
PSG               Ligue 1        Ángel Di María              7380005
Juventus          Serie A        Adrien Rabiot               7115707
Inter Milan       Serie A        Alexis Sánchez              7115707
AC Milan          Serie A        Zlatan Ibrahimovic          7115707
PSG               Ligue 1        Georginio Wijnaldum         7115707
Atletico Madrid   La Liga        João Félix                  7115707
Zenit             Premier Liga   Malcom                      7075046
Monaco            Ligue 1        Wissam Ben Yedder          

## 7.Рейтинг лиг по зп ведомости<a name="view3"></a>
Место в рейтинге - Лига	- Максимальная зп -  Перцентиль 90 -Перцентиль 80 -Перцентиль 60 -Перцентиль 40

In [66]:
liga = data_new.groupby(['league']).agg({'net_py':'max'})
liga.sort_values(by='net_py',ascending = False,inplace=True)
display(liga.shape)

(25, 1)

In [67]:
percent_40 = data_new.groupby('league')['net_py'].quantile(0.4)
percent_60 = data_new.groupby('league')['net_py'].quantile(0.6)
percent_80 = data_new.groupby('league')['net_py'].quantile(0.8)
percent_90 = data_new.groupby('league')['net_py'].quantile(0.9)

data_perc_ligue = pd.DataFrame()
data_perc_ligue['percentil40'] = percent_40
data_perc_ligue['percentil60'] = percent_60
data_perc_ligue['percentil80'] = percent_80 
data_perc_ligue['percentil90'] = percent_90
display(data_perc_ligue.shape)

(25, 4)

In [68]:
#соеденим таблицы 
rating_ligue = liga.merge(data_perc_ligue ,left_on='league' ,right_on = 'league' )
display(rating_ligue.shape)

(25, 5)

## 7.1 Итоговая таблица по рейтингу лиг<a name="zp1"></a>

In [69]:
#Рейтинг лиг по зп ведомостям
final_ligue = rating_ligue.rename({'league':'лига','net_py':'макс зп','percentil40':'перцентиль40','percentil60':'перцентиль60','percentil80':'перцентиль80','percentil90':'перцентиль90'}, axis=1)
final_ligue = final_ligue.sort_values(by=['макс зп'],ascending=False).reset_index()
final_ligue.groupby('league')
display(final_ligue.shape)
final_ligue



(25, 6)

league   макс зп  перцентиль40  перцентиль60  перцентиль80  \
0               Ligue 1  50826480     121984.00     274463.00     599752.00   
1        Premier League  30141507     882543.00    1724094.00    2760962.00   
2               La Liga  16589763     315124.00     630248.00    1463803.00   
3          1.Bundesliga  12391496     223637.00     453372.00     914877.00   
4               Serie A   8132237     406612.00     813224.00    1524794.00   
5          Premier Liga   7075046     304959.00     548926.00     853885.00   
6             Super Lig   5082648     172810.00     345620.00     677009.00   
7            Brasileiro   3900000      50000.00     150000.00     330000.00   
8               Liga MX   3760000     100000.00     210000.00     362000.00   
9                   MLS   3600000      79200.00     165216.00     330000.00   
10          Premiership   3556698      36170.00      48226.00      84396.00   
11         Championship   3448188     144679.00     253189.00     470208.00   
12        Primeira Liga   3049589      40661.00      60992.00     111818.00   
13              Serie B   2541324      81322.00     152479.00     280562.00   
14         2.Bundesliga   2429506     101653.00     162645.00     355785.00   
15           Eredivisie   2073720      40661.00      71157.00     162645.00   
16            La Liga 2   2033059      60992.00     101653.00     193141.00   
17     First Division A   1728100      60992.00     121984.00     203306.00   
18  Argentinian Primera   1300000     100000.00     160000.00     300000.00   
19         Super League   1189340      91488.00     142314.00     243967.00   
20           Bundesliga   1067356      40661.00      71157.00     132149.00   
21           League One    831906      48226.00      84396.00     120566.00   
22             Colombia    690000      30000.00      60000.00     100000.00   
23              Ligue 2    640414      30496.00      71157.00     121984.00   
24           League Two    168792      24113.00      36170.00      60283.00   

    перцентиль90  
0     1209670.00  
1     4135415.00  
2     3067886.00  
3     1597984.20  
4     2541324.00  
5     1242199.00  
6     1075488.00  
7      582000.00  
8      500000.00  
9      506398.80  
10     253189.00  
11     687226.00  
12     264298.00  
13     406612.00  
14     518430.00  
15     264298.00  
16     294794.00  
17     325289.00  
18     460000.00  
19     334438.40  
20     182975.00  
21     168792.00  
22     160000.00  
23     182975.00  
24      72942.80

In [70]:
mls = data_new.query('league == "MLS"')

In [71]:
mls

player   net_pw   net_py posotion  age  \
201                 Chicharito    69231  3600000        F   33   
266            Gonzalo Higuaín  58846,2  3060000        F   33   
320                Carlos Vela  51922,8  2700000        F   32   
405          Alejandro Pozuelo  43846,2  2280000        F   30   
438                Luiz Araújo  41538,6  2160000        F   25   
457             Josef Martínez  40384,8  2100000        F   28   
698          Sebastián Driussi    29670  1542852        F   25   
756               Maxi Moralez  28846,2  1500000        F   34   
781             Victor Wanyama  27692,4  1440000        M   30   
803               Adrien Hunou  27115,2  1410000        F   27   
843                Franco Jara  26422,8  1374000        F   33   
848               Robert Beric    26250  1364997        F   30   
866                 Carles Gil  25961,4  1350000        F   28   
891                Alan Pulido  25384,8  1320000        F   30   
892           Sebastián Blanco  25384,8  1320000        F   33   
904              Jozy Altidore  24937,8  1296750        F   31   
919              Albert Rusnák  24807,6  1290000        F   27   
925                Gustavo Bou  24461,4  1272000        F   31   
926            Nicolás Lodeiro  24461,4  1272000        F   32   
943               Raúl Ruidíaz    24231  1260000        F   31   
960             Gastón Giménez  23746,2  1234800        M   30   
1024            Luciano Acosta    22425  1166100        F   27   
1026                Ryan Gauld  22153,8  1152000        F   25   
1095          Mauricio Pereyra    20769  1080000        F   31   
1170          Darlington Nagbe    19731  1026000        M   31   
1257           Lucas Zelarayán  18692,4   972000        F   29   
1268             Edison Flores  18461,4   960000        F   27   
1269            Johnny Russell  18461,4   960000        F   31   
1307         Cecilio Domínguez  17653,8   918000        F   27   
1308           Rodolfo Pizarro  35192,4   915000        F   27   
1348              Hany Mukhtar  17365,2   903000        F   26   
1349            Blaise Matuidi  17307,6   900000        M   34   
1350           Michael Bradley  17307,6   900000        M   34   
1438              Kévin Cabral  16615,2   864000        F   22   
1530           Darwin Quintero  15600,6   811233        F   34   
1531              Gyasi Zardes  15577,2   810000        F   30   
1592           Jamiro Monteiro    15000   780000        M   27   
1597               Jürgen Damm  14884,8   774000        F   28   
1674                   Brenner  14098,2   733090        F   21   
1706            Damir Kreilach  13846,2   720000        M   32   
1709               Yimmi Chará  13846,2   720000        F   30   
1772                      Nani  26923,2   699999        F   34   
1777              Everton Luiz  13384,8   696000        M   33   
1793         Cristian Espinoza    13269   690000        F   26   
1795             Jordan Morris    13269   690000        F   27   
1827        Maximiliano Urruti  13109,4   681676        F   30   
1840             Andreu Fontàs    12981   675000        D   31   
1887              Aljaz Struna  12692,4   660000        D   31   
1889           Lucas Cavallini  12692,4   660000        F   28   
1891                 Yuya Kubo  12588,6   654600        F   27   
1918           Aké Arnaud Loba  12427,2   646200        F   23   
1996                Allan Cruz    11769   612000        M   25   
2071                Adam Buksa  11538,6   600000        F   25   
2072            Alexander Ring  11538,6   600000        M   30   
2073             Álvaro Medrán  11538,6   600000        M   27   
2080              Ilie Sánchez  11538,6   600000        M   30   
2081                João Paulo  11538,6   600000        F   30   
2083       Jonathan dos Santos  23077,2   600000        M   31   
2085           Jonathan Mensah  11538,6   600000        D   31   
2086             Kellyn Acosta  11538,6   600000        M   26  